sites/partners/src/listings/PaperListingForm/UnitsSummaryAmiForm.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { useEffect, useState } from "react"
+import {
+  t,
+  GridSection,
+  ViewItem,
+  GridCell,
+  Field,
+  Select,
+  AppearanceStyleType,
+  AppearanceBorderType,
+  FieldGroup,
+  Button,
+  Form,
+  SelectOption,
+} from "@bloom-housing/ui-components"
+import { useForm } from "react-hook-form"
+import { TempAmiLevel } from "./formTypes"
+import { AmiChart, MonthlyRentDeterminationType } from "@bloom-housing/backend-core/types"
+
+type UnitsSummaryFormProps = {
+  onSubmit: (amiLevel: TempAmiLevel) => void
+  onClose: () => void
+  amiLevels: TempAmiLevel[]
+  currentTempId: number
+  amiCharOptions: SelectOption[]
+  amiInfo: AmiChart[]
+}
+
+const UnitsSummaryAmiForm = ({
+  onSubmit,
+  onClose,
+  amiLevels,
+  currentTempId,
+  amiCharOptions,
+  amiInfo,
+}: UnitsSummaryFormProps) => {
+  const [current, setCurrent] = useState<TempAmiLevel>(null)
+  const [amiPercentageOptions, setAmiPercentageOptions] = useState<SelectOption[]>([])
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { register, errors, trigger, getValues, reset, watch } = useForm({
+    defaultValues: {
+      amiChartId: current?.amiChartId,
+      amiPercentage: current?.amiPercentage,
+      flatRentValue: current?.flatRentValue,
+      id: current?.id,
+      monthlyRentDeterminationType: current?.monthlyRentDeterminationType,
+      percentageOfIncomeValue: current?.percentageOfIncomeValue,
+      tempId: current?.tempId,
+    },
+  })
+  const monthlyRentDeterminationType: MonthlyRentDeterminationType = watch(
+    "monthlyRentDeterminationType"
+  )
+  const amiChartId: string = watch("amiChartId")
+
+  const fetchAmiChart = (chartId: string) => {
+    const chart = amiInfo.find((ami) => ami.id === chartId)
+    if (chart) {
+      setAmiPercentageOptions(
+        chart.items.reduce((accum, item) => {
+          if (!accum.some((percentage) => percentage.value === `${item.percentOfAmi}`)) {
+            accum.push({ value: `${item.percentOfAmi}`, label: `${item.percentOfAmi}` })
+          }
+          return accum
+        }, [] as SelectOption[])
+      )
+    }
+  }
+
+  useEffect(() => {
+    const amilevel = amiLevels.find((summary) => summary.tempId === currentTempId)
+    if (!amilevel?.amiChartId) {
+      setAmiPercentageOptions([])
+    } else {
+      fetchAmiChart(amilevel.amiChartId)
+    }
+    setCurrent(amilevel)
+    reset({ ...amilevel })
+  }, [])
```

Comment:
I don't think we should ignore the dependencies, which means `fetchAmiChart` needs to be wrapped in a useCallback.